spring-rest2ts-generator/src/main/java/com/blueveery/springrest2ts/converters/ModelClassesToTsClassesConverter.java

```diff
@@ -0,0 +1,127 @@
+package com.blueveery.springrest2ts.converters;
+
+import com.blueveery.springrest2ts.implgens.ImplementationGenerator;
+import com.blueveery.springrest2ts.naming.ClassNameMapper;
+import com.blueveery.springrest2ts.tsmodel.TSClass;
+import com.blueveery.springrest2ts.tsmodel.TSComplexElement;
+import com.blueveery.springrest2ts.tsmodel.TSField;
+import com.blueveery.springrest2ts.tsmodel.TSInterface;
+import com.blueveery.springrest2ts.tsmodel.TSModule;
+import com.blueveery.springrest2ts.tsmodel.TSType;
+import com.blueveery.springrest2ts.tsmodel.generics.IParameterizedWithFormalTypes;
+import com.blueveery.springrest2ts.tsmodel.generics.TSClassReference;
+import com.blueveery.springrest2ts.tsmodel.generics.TSInterfaceReference;
+import com.blueveery.springrest2ts.tsmodel.generics.TSParameterizedTypeReference;
+
+import java.lang.reflect.AnnotatedType;
+import java.util.List;
+import java.util.SortedSet;
+
+public class ModelClassesToTsClassesConverter extends ModelClassesAbstractConverter {
+
+    public ModelClassesToTsClassesConverter(
+            ImplementationGenerator implementationGenerator, ObjectMapper objectMapper
+    ) {
+        super(implementationGenerator, objectMapper);
+    }
+
+    public ModelClassesToTsClassesConverter(
+            ImplementationGenerator implementationGenerator, ClassNameMapper classNameMapper, ObjectMapper objectMapper
+    ) {
+        super(implementationGenerator, classNameMapper, objectMapper);
+    }
+
+    @Override
+    public boolean preConverted(JavaPackageToTsModuleConverter javaPackageToTsModuleConverter, Class javaClass) {
+        if (TypeMapper.map(javaClass) == TypeMapper.tsAny) {
+            ObjectMapper objectMapper = selectObjectMapper(javaClass);
+            if (objectMapper.filterClass(javaClass)) {
+                TSModule tsModule = javaPackageToTsModuleConverter.getTsModule(javaClass);
+                if (javaClass.isInterface()) {
+                    TSInterface tsInterface = new TSInterface(createTsClassName(javaClass), tsModule);
+                    tsModule.addScopedElement(tsInterface);
+                    TypeMapper.registerTsType(javaClass, tsInterface);
+                } else {
+                    TSClass tsClass = new TSClass(createTsClassName(javaClass), tsModule, getImplementationGenerator());
+                    tsModule.addScopedElement(tsClass);
+                    TypeMapper.registerTsType(javaClass, tsClass);
+                }
+                return true;
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public void convertInheritance(Class javaClass) {
```

Comment:
Raw use of parameterized class 'Class' shouldn't it be with wildcard?

Reply:
wildcard causes ClastCastException and requires larger refactoring, so the answer is NO for now